pkg/storage/helper/helper.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-2019 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package helper
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/pkg/errors"
+
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+)
+
+type layoutTemplate struct {
+	Username      string //the username
+	UsernameLower string //the username in lowercase
+	FirstLetter   string //first letter of username in lowercase
```

Comment:
Currently, `FirstLetter` limits that dir to a single rune. I'd like to see a variable length to use as the name for the parent dir. I lack the words for a better name of the `Parent` directory. Prefix does not cut it. Maybe `bucket`. For s3 storages we were considering the first n chars of an md5 / sha1 of the username to spread users among multiple buckets. But never followed up on it because randomly putting users into buckets seems nice, but in practice you'll end up with unevenly filled buckets when a few powerusers land in the same bucket. Which is why we stored a users home in the account table. That allows moving a single user to a different storage ...

This is better than the current code so still a +1 from me ;-)

Reply:
Good idea, added `{{.UsernamePrefixCount.x}}``